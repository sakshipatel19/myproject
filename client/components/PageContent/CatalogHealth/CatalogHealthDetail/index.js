import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import * as actions from "./cataloghealth-actions";
import { setSelectedPageInConfig } from "../../../global-actions";
import CatalogHealthDetail from "./CatalogHealthDeatil";

const mapStateToProps = (store, props) => {
	return {
		...props,
		...store.catalogDetail,
		categoryNames: store.global.config.categoryNames
	};
};

const mapDispatchToProps = dispatch => {
	return bindActionCreators({ ...actions, setSelectedPageInConfig }, dispatch);
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(CatalogHealthDetail);
