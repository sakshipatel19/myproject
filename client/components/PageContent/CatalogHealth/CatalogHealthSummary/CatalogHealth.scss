.catalog-health-container {
	padding-top: 30px;
	.label-container {
		color: #212129;
		font-family: "NotoSans-Bold";
		font-size: 22px;
		letter-spacing: 0px;
		line-height: 32px;
	}

	.catalog-health-list-container {
		display: flex;
		flex-direction: row;
		justify-content: space-between;
		margin-top: 20px;
		padding-left: 20px;

		.catalog-loading-placeholder {
			height: 65px;
			width: 200px;
			padding: 20px 10px;
		}
		.catalog-health-link {
			text-decoration: none;
			width: 100%;
			max-width: 290px;
			height: auto;
			margin-right: 30px;
		}
		.catalog-health-link:last-child {
			margin-right: -10px;
		}
		.catalog-health-list:hover {
			background-color: #ebf4fe;
			border-radius: 4px;
			.icon-right-arrow {
				margin-left: 8px;
			}
		}
		.catalog-health-list {
			padding: 10px 20px;
			font-family: "NotoSans-Regular";
			.catalog-health-list-heading {
				color: #212129;
				display: flex;
				font-size: 18px;
				letter-spacing: 0px;
				line-height: 26px;
				.arrow {
					margin-left: 10px;
				}
			}
			.catalog-health-list-item {
				display: flex;
				flex-direction: row;
				margin-top: 10px;
				.catalog-health-list-item-score {
					font-family: "NotoSans-Light";
					font-size: 28px;
					letter-spacing: 0px;
					line-height: 42px;
					margin-right: 15px;
				}
				.catalog-health-list-item-desc {
					font-family: "NotoSans-Regular";
					width: 70%;
					color: #212129;
					font-size: 14px;
					line-height: 20px;
					letter-spacing: 0px;
					padding-top: 5px;
				}
			}
		}
	}
}
