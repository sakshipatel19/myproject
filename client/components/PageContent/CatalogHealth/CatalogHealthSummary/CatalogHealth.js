import React from "react";

import { Link } from "react-router-dom";
import { withRouter } from "react-router";

import "./CatalogHealth.scss";
import Label from "../../../common/Label";
import Icon from "../../../common/Icon";
import DataLoadError from "../../common/DataLoadError/DataLoadError";
import LoadingSquare from "../../../common/LoadingSkeleton/LoadingSquare";

const CatalogHealth = props => {
  const { catalogHealthScore } = props;
  const CatalogHealthLists = catalogHealthScore?.data?.descriptions || [];
  const createCatalogHealthList = () => {
    return CatalogHealthLists.map((item, i) => {
      const colors = ["#00B0A3", "#FCA400", "#FF7A77"];
      return (
        <Link
          to={`${props.location.pathname}/catalog-health`}
          onClick={() => props.setSelectedTab(item?.key)}
          className={"catalog-health-link"}
          key={`catalog-health-link-${item?.descriptionType}_${i}`}
        >
          <div className="catalog-health-list">
            <div
              className="catalog-health-list-heading"
              key={`catalog-health-list-heading-${item?.descriptionType}`}
            >
              {item?.descriptionType}

              <span className="arrow">
                <Icon
                  name={props.arrowIconName || "sortRight"}
                  iconClass={"icon-right-arrow"}
                  size={12}
                />
              </span>
            </div>

            <div className="catalog-health-list-item">
              <div
                className="catalog-health-list-item-score"
                style={{ color: colors[i] }}
                key={`catalog-health-list-item-score-${item?.descriptionType}`}
              >{`${item?.descriptionScore}%`}</div>
              <div
                className="catalog-health-list-item-desc"
                key={`catalog-health-list-item-desc-${item?.descriptionType}`}
              >
                {item?.descriptionComment}
              </div>
            </div>
          </div>
        </Link>
      );
    });
  };
  const createCatalogLoading = [1, 2, 3].map(i => {
    return (
      <div
        className="catalog-loading-placeholder"
        key={`catalog-loading-placeholder-${i}`}
      >
        <LoadingSquare />
      </div>
    );
  });

  return (
    <div className="catalog-health-container">
      <Label text="Catalogue Health" labelClass="section-header-title" />
      <div className="catalog-health-list-container">
        {props?.catalogHealthScore?.fetching ? (
          <>{createCatalogLoading}</>
        ) : (
            <>
              {catalogHealthScore?.error ? (
                <DataLoadError
                  handleRetry={props.fetchCataloghealthScore}
                  error={catalogHealthScore.error}
                />
              ) : (
                  createCatalogHealthList()
                )}
            </>
          )}
      </div>
    </div>
  );
};

export default withRouter(CatalogHealth);
