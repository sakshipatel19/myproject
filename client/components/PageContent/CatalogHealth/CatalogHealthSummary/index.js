import CatalogHealth from "./CatalogHealth";
import {setSelectedTab} from "../CatalogHealthDetail/cataloghealth-actions";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
const mapDispatchToProps = dispatch =>
	bindActionCreators({ setSelectedTab }, dispatch);

export default connect(null, mapDispatchToProps)(CatalogHealth);
