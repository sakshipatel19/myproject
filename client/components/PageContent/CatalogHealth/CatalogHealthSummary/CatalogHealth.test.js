import React from "react";
import { shallow } from "enzyme";

import CatalogHealth from "./CatalogHealth";

describe("CatalogHealth tests", () => {
    let baseProps;
    beforeEach(() => {
        baseProps = {
            catalogHealthScore: {
                fetching: false,
                error: false,
                data: {
                    catalogHeading: "CATALOG HEALTH",
                    descriptions: [
                        {
                            descriptionType: "GOOD",
                            descriptionScore: "35",
                            descriptionComment: "xyz",
                            key: "good"
                        },
                        {
                            descriptionType: "ACCEPTABLE",
                            descriptionScore: "35",
                            descriptionComment: "xyz",
                            key: "acceptable"
                        },
                        {
                            descriptionType: "NEEDS IMPROVEMENT",
                            descriptionScore: "35",
                            descriptionComment: "xyz",
                            key: "needsAttention"
                        },
                    ]
                }
            },
            fetchCataloghealthScore: jest.fn(),
            location: {
                pathname: '/analysis/catalog-health'
            }
        };
    });
    it("CatalogHealth should render correctly", () => {
        const wrapper = shallow(<CatalogHealth.WrappedComponent {...baseProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
