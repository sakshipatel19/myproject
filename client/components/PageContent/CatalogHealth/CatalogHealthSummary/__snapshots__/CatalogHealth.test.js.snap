// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogHealth tests CatalogHealth should render correctly 1`] = `
<div
  className="catalog-health-container"
>
  <Label
    labelClass="section-header-title"
    text="Catalogue Health"
  />
  <div
    className="catalog-health-list-container"
  >
    <Link
      className="catalog-health-link"
      key="catalog-health-link-GOOD_0"
      onClick={[Function]}
      to="/analysis/catalog-health/catalog-health"
    >
      <div
        className="catalog-health-list"
      >
        <div
          className="catalog-health-list-heading"
          key="catalog-health-list-heading-GOOD"
        >
          GOOD
          <span
            className="arrow"
          >
            <Icon
              iconClass="icon-right-arrow"
              name="sortRight"
              size={12}
            />
          </span>
        </div>
        <div
          className="catalog-health-list-item"
        >
          <div
            className="catalog-health-list-item-score"
            key="catalog-health-list-item-score-GOOD"
            style={
              Object {
                "color": "#00B0A3",
              }
            }
          >
            35%
          </div>
          <div
            className="catalog-health-list-item-desc"
            key="catalog-health-list-item-desc-GOOD"
          >
            xyz
          </div>
        </div>
      </div>
    </Link>
    <Link
      className="catalog-health-link"
      key="catalog-health-link-ACCEPTABLE_1"
      onClick={[Function]}
      to="/analysis/catalog-health/catalog-health"
    >
      <div
        className="catalog-health-list"
      >
        <div
          className="catalog-health-list-heading"
          key="catalog-health-list-heading-ACCEPTABLE"
        >
          ACCEPTABLE
          <span
            className="arrow"
          >
            <Icon
              iconClass="icon-right-arrow"
              name="sortRight"
              size={12}
            />
          </span>
        </div>
        <div
          className="catalog-health-list-item"
        >
          <div
            className="catalog-health-list-item-score"
            key="catalog-health-list-item-score-ACCEPTABLE"
            style={
              Object {
                "color": "#FCA400",
              }
            }
          >
            35%
          </div>
          <div
            className="catalog-health-list-item-desc"
            key="catalog-health-list-item-desc-ACCEPTABLE"
          >
            xyz
          </div>
        </div>
      </div>
    </Link>
    <Link
      className="catalog-health-link"
      key="catalog-health-link-NEEDS IMPROVEMENT_2"
      onClick={[Function]}
      to="/analysis/catalog-health/catalog-health"
    >
      <div
        className="catalog-health-list"
      >
        <div
          className="catalog-health-list-heading"
          key="catalog-health-list-heading-NEEDS IMPROVEMENT"
        >
          NEEDS IMPROVEMENT
          <span
            className="arrow"
          >
            <Icon
              iconClass="icon-right-arrow"
              name="sortRight"
              size={12}
            />
          </span>
        </div>
        <div
          className="catalog-health-list-item"
        >
          <div
            className="catalog-health-list-item-score"
            key="catalog-health-list-item-score-NEEDS IMPROVEMENT"
            style={
              Object {
                "color": "#FF7A77",
              }
            }
          >
            35%
          </div>
          <div
            className="catalog-health-list-item-desc"
            key="catalog-health-list-item-desc-NEEDS IMPROVEMENT"
          >
            xyz
          </div>
        </div>
      </div>
    </Link>
  </div>
</div>
`;
